src/main/java/seedu/weme/statistics/TagManager.java

```diff
@@ -0,0 +1,100 @@
+package seedu.weme.statistics;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Set;
+
+import javafx.collections.ObservableList;
+
+import seedu.weme.model.ReadOnlyMemeBook;
+import seedu.weme.model.meme.Meme;
+import seedu.weme.model.tag.Tag;
+
+/**
+ * Tag manager in Statistics feature that reads tags from a {@code MemeBook}.
+ */
+public class TagManager {
+
+    private Set<Tag> tags;
+    private PriorityQueue<TagWithCount> tagsWithCount;
+
+    public TagManager() {
+        tags = new HashSet<>();
+        tagsWithCount = new PriorityQueue<>();
+    }
+
+    public Set<Tag> getTagsInSet() {
+        return tags;
+    }
+
+    /**
+     * Get {@code TagWithCount} in PriorityQueue from the pre-stored data.
+     */
+    public PriorityQueue<TagWithCount> getTagsWithCountInPriorityQ() {
```

Comment:
any reason for having both this method and the one below?

Reply:
The one without argument fetches the stored PQ, while the one that takes in argument can overwrite the current data with the input and generate PQ again.